docs/user/installing-on-vm.rst

```diff
@@ -0,0 +1,138 @@
+Install OpenWISP for Testing in a VirtualBox VM
+===============================================
+
+If you want to try out OpenWISP in your own development environment, the
+safest way is to use a VirtualBox Virtual Machine (from here on VM).
+
+Using Vagrant
+-------------
+
+**Since August 2018 there's a new fast and easy way to install OpenWISP
+for testing purposes** leveraging `Vagrant <https://www.vagrantup.com>`__,
+a popular open source tool for building and maintaining portable virtual
+software development environments.
+
+To use this new way, clone the repository `vagrant-openwisp2
+<https://github.com/openwisp/vagrant-openwisp2>`__, it contains the
+instructions (in the ``README.md``) and the vagrant configuration to
+perform the automatic installation.
+
+Alternatively, you can read on to learn how to install *VirtualBox* and
+run *ansible-openwisp2* manually, this is useful if you need to test
+advanced customizations of *OpenWISP*.
+
+Installing Debian 11 on VirtualBox
+----------------------------------
+
+Install `VirtualBox <https://virtualbox.org>`__ and create a new Virtual
+Machine running Debian 11. A step-by-step guide is available `here
+<http://www.brianlinkletter.com/installing-debian-linux-in-a-virtualbox-virtual-machine/>`__,
+however we need to change a few things to get ansible working.
+
+VM Configuration
+----------------
+
+Proceed with the installation as shown in the guide linked above, and come
+back here when you see this screen:
+
+.. figure:: https://raw.githubusercontent.com/openwisp/ansible-openwisp2/master/docs/debian-software-selection.png
```

Comment:
For consistency, let's move all the images to a dedicated directory eg:  `docs/images/`.